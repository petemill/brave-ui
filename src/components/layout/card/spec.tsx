/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Card from './index'

describe('card tests', () => {
  const baseComponent = (props?: object) => (
    <Card id='testCard' {...props} />
  )
  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent({id: 'toughCard'}))
      const assertion = wrapper.find('#toughCard').length
      expect(assertion).toBe(1)
    })
  })

  describe('component behavior', () => {
    it('can have an id', () => {
      const wrapper = shallow(baseComponent({id: 'boxedThing'}))
      const assertion = wrapper.props().id
      expect(assertion).toBe('boxedThing')
    })
  })
})
