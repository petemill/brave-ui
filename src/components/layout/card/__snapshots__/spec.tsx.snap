// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card tests basic tests matches the snapshot 1`] = `
.c0 {
  padding: 40px;
  background-color: #FFF;
  border-radius: 6px;
  box-shadow: 0 0 8px 0 rgba(99,105,110,0.12);
  font-size: inherit;
  box-sizing: border-box;
  position: relative;
}

<div
  className="c0"
  id="testCard"
/>
`;
